### PR TITLE
Use capsh instead of setcap to start pihole-FTL

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:** 
*please fill any appropriate checkboxes, e.g: [X]*

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes, and have included unit tests where possible.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

Please make sure you [Sign Off](https://github.com/pi-hole/pi-hole/wiki/How-to-signoff-your-commits.) all commits. Pi-hole enforces the [DCO](https://github.com/pi-hole/pi-hole/wiki/Contributing-to-the-project).

---
**What does this PR aim to accomplish?:**

Change from using `setcap` to `capsh`.

This is done for two reasons:

1. Use `capsh` instead of `setcap` to allow running `pihole-FTL` as non-privileged user on filesystems that don't support capabilities.
2. Fixing a security issue. When using `setcap`, the binary is hot so that any user can now call it and take priv'd ports. When using `capsh` only `root` has that ability and it's only done during the `/etc/init.d/` calls, so the binary itself is just as normal as any other binary on the system.

**How does this PR accomplish the above?:**

Change from using `setcap` to `capsh`.

**What documentation changes (if any) are needed to support this PR?:**

None

### CODE REVIEW

**Comment by Mcat12** [comment]
The CI builds should be fixed by #2672

File: advanced/Templates/pihole-FTL.service [hunk 1]
```diff
@@ -42,10 +43,11 @@ start() {
     chown pihole:pihole /var/log/pihole-FTL.log /var/log/pihole.log
     chmod 0644 /var/log/pihole-FTL.log /run/pihole-FTL.pid /run/pihole-FTL.port /var/log/pihole.log
     echo "nameserver 127.0.0.1" | /sbin/resolvconf -a lo.piholeFTL
-    if setcap CAP_NET_BIND_SERVICE,CAP_NET_RAW,CAP_NET_ADMIN+eip "$(which pihole-FTL)"; then
-      su -s /bin/sh -c "/usr/bin/pihole-FTL" "$FTLUSER"
-    else
-      echo "Warning: Starting pihole-FTL as root because setting capabilities is not supported on this system"
+    # capsh --keep=1 --secbits=0x04 enables SECBIT_NO_SETUID_FIXUP preventing the kernel from
+    # stripping capabilities when changing from UID=0 to UID=nonzero
+    capsh --keep=1 --secbits=0x04 --caps="CAP_NET_BIND_SERVICE,CAP_NET_RAW,CAP_NET_ADMIN+eip" -- -c "runuser -u ${FTLUSER} ${FTLBINARY}"
```
Comment (Mcat12) [inline advanced/Templates/pihole-FTL.service]: There appears to be a `--user` option for `capsh`, is that portable?

Comment (DL6ER) [reply advanced/Templates/pihole-FTL.service]: It doesn't seem to work as expected.
```
root@vultr:~# capsh --keep=1 --secbits=0x04 --caps="CAP_NET_BIND_SERVICE,CAP_NET_RAW,CAP_NET_ADMIN+eip" --user="pihole" -c "/usr/bin/pihole-FTL"
Unable to set group list for user: Operation not permitted
```

**Comment by DL6ER** [comment]
The current set of capabilities added by `capsh` is insufficient. See output of the new `DEBUG_CAPS` flag:
```
[2019-03-18 17:08:31.031 24520] ***************************************
[2019-03-18 17:08:31.031 24520] * Linux capability debugging enabled  *
[2019-03-18 17:08:31.031 24520] * CAP_CHOWN                (00) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_DAC_OVERRIDE         (01) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_DAC_READ_SEARCH      (02) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_FOWNER               (03) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_FSETID               (04) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_KILL                 (05) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_SETGID               (06) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_SETUID               (07) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_SETPCAP              (08) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_LINUX_IMMUTABLE      (09) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_NET_BIND_SERVICE     (10) = -I- *
[2019-03-18 17:08:31.031 24520] * CAP_NET_BROADCAST        (11) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_NET_ADMIN            (12) = -I- *
[2019-03-18 17:08:31.031 24520] * CAP_NET_RAW              (13) = -I- *
[2019-03-18 17:08:31.031 24520] * CAP_IPC_LOCK             (14) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_IPC_OWNER            (15) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_SYS_MODULE           (16) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_SYS_RAWIO            (17) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_SYS_CHROOT           (18) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_SYS_PTRACE           (19) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_SYS_PACCT            (20) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_SYS_ADMIN            (21) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_SYS_BOOT             (22) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_SYS_NICE             (23) = --- *
[2019-03-18 17:08:31.031 24520] * CAP_SYS_RESOURCE         (24) = --- *
[2019-03-18 17:08:31.033 24520] * CAP_SYS_TIME             (25) = --- *
[2019-03-18 17:08:31.033 24520] * CAP_SYS_TTY_CONFIG       (26) = --- *
[2019-03-18 17:08:31.033 24520] * CAP_MKNOD                (27) = --- *
[2019-03-18 17:08:31.033 24520] * CAP_LEASE                (28) = --- *
[2019-03-18 17:08:31.033 24520] * CAP_AUDIT_WRITE          (29) = --- *
[2019-03-18 17:08:31.033 24520] * CAP_AUDIT_CONTROL        (30) = --- *
[2019-03-18 17:08:31.033 24520] * CAP_SETFCAP              (31) = --- *
[2019-03-18 17:08:31.033 24520] * CAP_MAC_OVERRIDE         (32) = --- *
[2019-03-18 17:08:31.033 24520] * CAP_MAC_ADMIN            (33) = --- *
[2019-03-18 17:08:31.033 24520] * CAP_SYSLOG               (34) = --- *
[2019-03-18 17:08:31.033 24520] * CAP_WAKE_ALARM           (35) = --- *
[2019-03-18 17:08:31.033 24520] * CAP_BLOCK_SUSPEND        (36) = --- *
[2019-03-18 17:08:31.033 24520] * CAP_AUDIT_READ           (37) = --- *
[2019-03-18 17:08:31.033 24520] ***************************************
[2019-03-18 17:08:31.033 24520] *************************************************************************
[2019-03-18 17:08:31.033 24520] * WARNING: Required Linux capability CAP_NET_ADMIN not available        *
[2019-03-18 17:08:31.033 24520] *************************************************************************
[2019-03-18 17:08:31.033 24520] *************************************************************************
[2019-03-18 17:08:31.033 24520] * WARNING: Required Linux capability CAP_NET_RAW not available          *
[2019-03-18 17:08:31.033 24520] *************************************************************************
[2019-03-18 17:08:31.033 24520] *************************************************************************
[2019-03-18 17:08:31.033 24520] * WARNING: Required Linux capability CAP_NET_BIND_SERVICE not available *
[2019-03-18 17:08:31.033 24520] *************************************************************************

dnsmasq: failed to create listening socket for port 53: Permission denied
```

File: advanced/Templates/pihole-FTL.service (hunk 1 above)
Comment (DL6ER) [reply advanced/Templates/pihole-FTL.service]: Okay, for this to work we have to add `CAP_SETUID,CAP_SETGID`

**Comment by DL6ER** [comment]
This PR, unfortunately, does not have a realistic chance to work as expected.

**Pull request closed without merging**